`sqlformat` cannot handle Unicode characters

Here is the SQL contains Unicode characters:

```sql
SELECT *
                FROM User
        WHERE name = '大圣'
```

After `sqlformat` the characters are scrambled:

```bash
$ sqlformat dump.sql --reindent
SELECT *
FROM User
WHERE name = 'å¤§å£'
```

I installed `sqlformat` by `pip install sqlparse`, and version is `0.2.2`:

```bash
$ sqlformat --version
0.2.2

@phdru oh, I'm sorry, it is from previous version.

I'm still trying to understand why `unicode-escape`, if I'm right this time, it's introduced in PR https://github.com/andialbrecht/sqlparse/pull/88 

```diff
-            text = text.decode(self.encoding)
+            try:
+                text = text.decode(self.encoding)
+            except UnicodeDecodeError:
+                text = text.decode('unicode-escape')
```

And the test case was added in commit https://github.com/andialbrecht/sqlparse/commit/7088a29fc330dd806e56fb4f7af0f9389bcb114a

```python
def test_parse_sql_with_binary():
    # See https://github.com/andialbrecht/sqlparse/pull/88
    digest = '\x82|\xcb\x0e\xea\x8aplL4\xa1h\x91\xf8N{'
    sql = "select * from foo where bar = '{0}'".format(digest)
    formatted = sqlparse.format(sql, reindent=True)
    tformatted = "select *\nfrom foo\nwhere bar = '{0}'".format(digest)
    if PY2:
        tformatted = tformatted.decode('unicode-escape')
    assert formatted == tformatted
```

By replacing `unicode-escape` with `utf-8` in `lexer.py` will cause this test case fail.

```bash
input = "select * from foo where bar = '\x82|\xcb\x0e\xea\x8aplL4\xa1h\x91\xf8N{'", errors = 'strict'

    def decode(input, errors='strict'):
>       return codecs.utf_8_decode(input, errors, True)
E       UnicodeDecodeError: 'utf8' codec can't decode byte 0x82 in position 31: invalid start byte

.tox/py27/lib/python2.7/encodings/utf_8.py:16: UnicodeDecodeError
```

After analyze the string `digest = '\x82|\xcb\x0e\xea\x8aplL4\xa1h\x91\xf8N{'`, I understand the reason why it raised the error. The `'\x82|\xcb\x0e\xea\x8aplL4\xa1h\x91\xf8N{'` is not `utf-8` encoded string, it's in `ISO 8859-1` encoding:

![2017-01-11 10 37 30](https://cloud.githubusercontent.com/assets/6299096/21829526/90af31c0-d7ea-11e6-91e5-365fc20945bb.png)


And Atom can verify that:

![2017-01-11 10 15 56](https://cloud.githubusercontent.com/assets/6299096/21828950/93a90e26-d7e7-11e6-91b4-ba4779e55e5d.png)

That is the reason why `utf-8` cannot decode `\x82`, which is indeed an invalid `utf-8` starting byte.

And I don't think `unicode-escape` encoding in the test case is correct, it should be `iso-8859-1`. The test case will pass if I change it this way:

```diff
     digest = '\x82|\xcb\x0e\xea\x8aplL4\xa1h\x91\xf8N{'
     sql = "select * from foo where bar = '{0}'".format(digest)
-    formatted = sqlparse.format(sql, reindent=True)
+    formatted = sqlparse.format(sql, reindent=True, encoding='iso-8859-1')
     tformatted = "select *\nfrom foo\nwhere bar = '{0}'".format(digest)
     if PY2:
-        tformatted = tformatted.decode('unicode-escape')

+        tformatted = tformatted.decode('iso-8859-1')
     assert formatted == tformatted
```

I think the default encoding should be `utf-8`, which is common way to solve encoding problem, and any other encoding should be specified, such as `encoding='iso-8859-1'` here. And in this case, a CLI option for encoding is necessary.

--- a/sqlparse/lexer.py
+++ b/sqlparse/lexer.py
@@ -44,7 +44,7 @@ class Lexer(object):
             pass
         elif isinstance(text, bytes_type):
             try:
-                text = text.decode()
+                text = text.decode('utf-8')
             except UnicodeDecodeError:
                 if not encoding:
                     encoding = 'unicode-escape'

--- a/sqlparse/lexer.py
+++ b/sqlparse/lexer.py
@@ -43,12 +43,13 @@ class Lexer(object):
         if isinstance(text, text_type):
             pass
         elif isinstance(text, bytes_type):
-            try:
-                text = text.decode('utf-8')
-            except UnicodeDecodeError:
-                if not encoding:
-                    encoding = 'unicode-escape'
+            if encoding:
                 text = text.decode(encoding)
+            else:
+                try:
+                    text = text.decode('utf-8')
+                except UnicodeDecodeError:
+                    text = text.decode('unicode-escape')
         else:
             raise TypeError(u"Expected text or file-like object, got {!r}".
                             format(type(text)))

--- a/tests/test_regressions.py
+++ b/tests/test_regressions.py
@@ -323,3 +323,22 @@ def test_token_next_doesnt_ignore_skip_cm():
 def test_issue284_as_grouping(s):
     p = sqlparse.parse(s)[0]
     assert s == str(p)
+
+
+def test_issue315_utf8_by_default():
+    # Make sure the lexer can handle utf-8 string by default correctly
+    # digest = '齐天大圣.カラフルな雲.사랑해요'
+    # The digest contains Chinese, Japanese and Korean characters
+    # All in 'utf-8' encoding.
+    digest = (
+        '\xe9\xbd\x90\xe5\xa4\xa9\xe5\xa4\xa7\xe5\x9c\xa3.'
+        '\xe3\x82\xab\xe3\x83\xa9\xe3\x83\x95\xe3\x83\xab\xe3\x81\xaa\xe9'
+        '\x9b\xb2.'
+        '\xec\x82\xac\xeb\x9e\x91\xed\x95\xb4\xec\x9a\x94'
+    )
+    sql = "select * from foo where bar = '{0}'".format(digest)
+    formatted = sqlparse.format(sql, reindent=True)
+    tformatted = "select *\nfrom foo\nwhere bar = '{0}'".format(digest)
+    if PY2:
+        tformatted = tformatted.decode('utf-8')
+    assert formatted == tformatted

